### ISSUE TITLE
[Bug]: panic: runtime error: index out of range [0] with length 0

### ISSUE THREAD

**Issue by sukki37** [matrixorigin/matrixone#2186]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

when executing `SELECT qq.Carrier, c, c2, c*100/c2 as c3 FROM (SELECT Carrier, count(*) AS c FROM ontime WHERE DepDelay>10 AND Year=1987 GROUP BY Carrier) qq join (SELECT Carrier, count(*) AS c2 FROM ontime WHERE Year=1987 GROUP BY Carrier)q on q.Carrier=qq.Carrier ORDER BY c3 DESC;` panic occurred

```
panic: runtime error: index out of range [0] with length 0
[signal SIGSEGV: segmentation violation code=0xc00abf3b70 addr=0x20af318 pc=0x1fe7a3e]
goroutine 15358 [running]:
github.com/matrixorigin/matrixone/pkg/sql/viewexec/join.Prepare(0x4, {0x23eb360, 0xc009c9c230})
        /mnt/datadisk0/demo/matrixone/pkg/sql/viewexec/join/join.go:77 +0xafd
github.com/matrixorigin/matrixone/pkg/vm.Prepare({0xc003349b00, 0x5, 0x18}, 0x7faace8411d8)
        /mnt/datadisk0/demo/matrixone/pkg/vm/vm.go:37 +0x8a
github.com/matrixorigin/matrixone/pkg/vm/pipeline.(*Pipeline).RunMerge(0xc00abf3f40, 0xc002fe3b00)
        /mnt/datadisk0/demo/matrixone/pkg/vm/pipeline/pipeline.go:109 +0x9f
github.com/matrixorigin/matrixone/pkg/sql/compile.(*Scope).MergeRun(0xc009bf2e00, {0x2a2bd20, 0xc003c64ec0})
        /mnt/datadisk0/demo/matrixone/pkg/sql/compile/scope.go:500 +0x4aa
github.com/matrixorigin/matrixone/pkg/sql/compile.(*Scope).MergeRun.func2(0x0)
        /mnt/datadisk0/demo/matrixone/pkg/sql/compile/scope.go:481 +0x2a
created by github.com/matrixorigin/matrixone/pkg/sql/compile.(*Scope).MergeRun
        /mnt/datadisk0/demo/matrixone/pkg/sql/compile/scope.go:480 +0x252
```

### Steps to Reproduce

```markdown
Table schema:

CREATE TABLE `ontime` (
  `Year` int DEFAULT 0,
  `Quarter` tinyint(4) DEFAULT NULL,
  `Month` tinyint(4) DEFAULT NULL,
  `DayofMonth` tinyint(4) DEFAULT NULL,
  `DayOfWeek` tinyint(4) DEFAULT NULL,
  `FlightDate` date DEFAULT NULL,
  `UniqueCarrier` char(7) DEFAULT NULL,
  `AirlineID` int(11) DEFAULT NULL,
  `Carrier` char(2) DEFAULT NULL,
  `TailNum` varchar(50) DEFAULT NULL,
  `FlightNum` varchar(10) DEFAULT NULL,
  `OriginAirportID` int(11) DEFAULT NULL,
  `OriginAirportSeqID` int(11) DEFAULT NULL,
  `OriginCityMarketID` int(11) DEFAULT NULL,
  `Origin` char(5) DEFAULT NULL,
  `OriginCityName` varchar(100) DEFAULT NULL,
  `OriginState` char(2) DEFAULT NULL,
  `OriginStateFips` varchar(10) DEFAULT NULL,
  `OriginStateName` varchar(100) DEFAULT NULL,
  `OriginWac` int(11) DEFAULT NULL,
  `DestAirportID` int(11) DEFAULT NULL,
  `DestAirportSeqID` int(11) DEFAULT NULL,
  `DestCityMarketID` int(11) DEFAULT NULL,
  `Dest` char(5) DEFAULT NULL,
  `DestCityName` varchar(100) DEFAULT NULL,
  `DestState` char(2) DEFAULT NULL,
  `DestStateFips` varchar(10) DEFAULT NULL,
  `DestStateName` varchar(100) DEFAULT NULL,
  `DestWac` int(11) DEFAULT NULL,
  `CRSDepTime` int(11) DEFAULT NULL,
  `DepTime` int(11) DEFAULT NULL,
  `DepDelay` float DEFAULT NULL,
  `DepDelayMinutes` float DEFAULT NULL,
  `DepDel15` float DEFAULT NULL,
  `DepartureDelayGroups` int(11) DEFAULT NULL,
  `DepTimeBlk` varchar(20) DEFAULT NULL,
  `TaxiOut` float DEFAULT NULL,
  `WheelsOff` int(11) DEFAULT NULL,
  `WheelsOn` int(11) DEFAULT NULL,
  `TaxiIn` float DEFAULT NULL,
  `CRSArrTime` int(11) DEFAULT NULL,
  `ArrTime` int(11) DEFAULT NULL,
  `ArrDelay` float DEFAULT NULL,
  `ArrDelayMinutes` float DEFAULT NULL,
  `ArrDel15` float DEFAULT NULL,
  `ArrivalDelayGroups` int(11) DEFAULT NULL,
  `ArrTimeBlk` varchar(20) DEFAULT NULL,
  `Cancelled` float DEFAULT NULL,
  `CancellationCode` char(1) DEFAULT NULL,
  `Diverted` float DEFAULT NULL,
  `CRSElapsedTime` float DEFAULT NULL,
  `ActualElapsedTime` float DEFAULT NULL,
  `AirTime` float DEFAULT NULL,
  `Flights` float DEFAULT NULL,
  `Distance` float DEFAULT NULL,
  `DistanceGroup` tinyint(4) DEFAULT NULL,
  `CarrierDelay` float DEFAULT NULL,
  `WeatherDelay` float DEFAULT NULL,
  `NASDelay` float DEFAULT NULL,
  `SecurityDelay` float DEFAULT NULL,
  `LateAircraftDelay` float DEFAULT NULL,
  `FirstDepTime` varchar(10) DEFAULT NULL,
  `TotalAddGTime` varchar(10) DEFAULT NULL,
  `LongestAddGTime` varchar(10) DEFAULT NULL,
  `DivAirportLandings` varchar(10) DEFAULT NULL,
  `DivReachedDest` varchar(10) DEFAULT NULL,
  `DivActualElapsedTime` varchar(10) DEFAULT NULL,
  `DivArrDelay` varchar(10) DEFAULT NULL,
  `DivDistance` varchar(10) DEFAULT NULL,
  `Div1Airport` varchar(10) DEFAULT NULL,
  `Div1AirportID` int(11) DEFAULT NULL,
  `Div1AirportSeqID` int(11) DEFAULT NULL,
  `Div1WheelsOn` varchar(10) DEFAULT NULL,
  `Div1TotalGTime` varchar(10) DEFAULT NULL,
  `Div1LongestGTime` varchar(10) DEFAULT NULL,
  `Div1WheelsOff` varchar(10) DEFAULT NULL,
  `Div1TailNum` varchar(10) DEFAULT NULL,
  `Div2Airport` varchar(10) DEFAULT NULL,
  `Div2AirportID` int(11) DEFAULT NULL,
  `Div2AirportSeqID` int(11) DEFAULT NULL,
  `Div2WheelsOn` varchar(10) DEFAULT NULL,
  `Div2TotalGTime` varchar(10) DEFAULT NULL,
  `Div2LongestGTime` varchar(10) DEFAULT NULL,
  `Div2WheelsOff` varchar(10) DEFAULT NULL,
  `Div2TailNum` varchar(10) DEFAULT NULL,
  `Div3Airport` varchar(10) DEFAULT NULL,
  `Div3AirportID` int(11) DEFAULT NULL,
  `Div3AirportSeqID` int(11) DEFAULT NULL,
  `Div3WheelsOn` varchar(10) DEFAULT NULL,
  `Div3TotalGTime` varchar(10) DEFAULT NULL,
  `Div3LongestGTime` varchar(10) DEFAULT NULL,
  `Div3WheelsOff` varchar(10) DEFAULT NULL,
  `Div3TailNum` varchar(10) DEFAULT NULL,
  `Div4Airport` varchar(10) DEFAULT NULL,
  `Div4AirportID` int(11) DEFAULT NULL,
  `Div4AirportSeqID` int(11) DEFAULT NULL,
  `Div4WheelsOn` varchar(10) DEFAULT NULL,
  `Div4TotalGTime` varchar(10) DEFAULT NULL,
  `Div4LongestGTime` varchar(10) DEFAULT NULL,
  `Div4WheelsOff` varchar(10) DEFAULT NULL,
  `Div4TailNum` varchar(10) DEFAULT NULL,
  `Div5Airport` varchar(10) DEFAULT NULL,
  `Div5AirportID` int(11) DEFAULT NULL,
  `Div5AirportSeqID` int(11) DEFAULT NULL,
  `Div5WheelsOn` varchar(10) DEFAULT NULL,
  `Div5TotalGTime` varchar(10) DEFAULT NULL,
  `Div5LongestGTime` varchar(10) DEFAULT NULL,
  `Div5WheelsOff` varchar(10) DEFAULT NULL,
  `Div5TailNum` varchar(10) DEFAULT NULL
);

load data infile 'ontime.csv' into table ontime FIELDS TERMINATED BY ',' OPTIONALLY ENCLOSED BY '"' LINES TERMINATED BY '\n' IGNORE 1 LINES;
```
```


### Additional information

_No response_

**Comment by sukki37** [comment]
Data is in the attachment.
[On_Time_Reporting_Carrier_On_Time_Performance_1987_present_1987_10.zip](https://github.com/matrixorigin/matrixone/files/8463424/On_Time_Reporting_Carrier_On_Time_Performance_1987_present_1987_10.zip)
.

**Comment by nnsgmsone** [comment]
fixed by #2199

**Comment by sukki37** [comment]
test passed
